### PR TITLE
Mitigate long branches issues on application restarts

### PR DESCRIPTION
**Pull request by MarekM25**

### CODE REVIEW

File: src/Nethermind/Nethermind.Blockchain/Visitors/StartupBlockTreeFixer.cs [hunk 1]
```diff
@@ -172,10 +241,30 @@ private void AssertNotVisitingAfterGap()
...
+                    $"Not expecting to visit block at {_currentLevelNumber} because the gap has already been identified.");
+            }
+        }
+
+        private bool CanSuggestBlocks(Block block)
+        {
+            _firstBlockVisited = false;
+            if (block?.ParentHash != null)
+            {
+                Block parentBlock = _blockTree.FindBlock(block.ParentHash,
```
Comment (tkstanczak) [inline src/Nethermind/Nethermind.Blockchain/Visitors/StartupBlockTreeFixer.cs:254]: parentBlock.FindParent blockw ould be much more efficient here?

Comment (tkstanczak) [reply src/Nethermind/Nethermind.Blockchain/Visitors/StartupBlockTreeFixer.cs:254]: public static BlockHeader? FindParentHeader(this IBlockFinder finder, BlockHeader header, BlockTreeLookupOptions options)

**Review by tkstanczak — CHANGES_REQUESTED** [review]
this would be better, I think: public static BlockHeader? FindParentHeader(this IBlockFinder finder, BlockHeader header, BlockTreeLookupOptions options)

File: src/Nethermind/Nethermind.Blockchain/Visitors/StartupBlockTreeFixer.cs [hunk 2]
```diff
@@ -43,26 +47,57 @@ public class StartupBlockTreeFixer : IBlockTreeVisitor
...
         private long? _processingGapStart;
 
-        public StartupBlockTreeFixer(ISyncConfig syncConfig, IBlockTree blockTree, ILogger logger)
+        private TaskCompletionSource _dbBatchProcessed;
+        private long _currentDbLoadBatchEnd;
+        private bool _firstBlockVisited = true;
+        private bool _suggestBlocks = true;
+        private readonly long _batchSize;
+
+        public StartupBlockTreeFixer(
```
Comment (LukaszRozmej) [inline src/Nethermind/Nethermind.Blockchain/Visitors/StartupBlockTreeFixer.cs:56]: couldn't we reuse DbBlocksLoader in some way?

**Review by LukaszRozmej — APPROVED** [review]

**Pull request merged**